crates/eth-rpc/tests/kkrt_prep.rs

```diff
@@ -0,0 +1,377 @@
+
+        to: TransactionKind::Create,
+
+        value: 0u128, // U256 used for large numbers
+
+        input: contract_bytes_reth,
+
+        access_list: Default::default(), // empty access list
+    });
+
+    let signature = sign_message(eoa_secret, transaction.signature_hash()).unwrap();
+
+    let signed_transaction = TransactionSigned::from_transaction_and_signature(transaction, signature);
+    let mut buffer = BytesMut::new(); // Create a new empty buffer
+
+    signed_transaction.encode_enveloped(&mut buffer); // Encode the transaction into the buffer
+    let bytes_vec = buffer.to_vec(); // Get Vec<u8> from Bytes
+
+    let field_elements: Vec<FieldElement> = bytes_vec
+        .into_iter()
+        .map(|byte| {
+            // Convert each byte to a field element
+            FieldElement::from(byte)
+        })
+        .collect();
+
+    let not_used_eoa_account_fields = FieldElement::from_hex_be("0xDEAD").unwrap();
+    let deployment_of_counter_evm_contract_result = eoa_starknet_account
+        .execute(vec![Call {
+            calldata: field_elements,
+            to: not_used_eoa_account_fields,
+            selector: not_used_eoa_account_fields,
+        }])
+        .send()
+        .await
+        .unwrap();
+
+    let deployment_of_counter_evm_contract_result_receipt = eoa_starknet_account
+        .provider()
+        .get_transaction_receipt(deployment_of_counter_evm_contract_result.transaction_hash)
+        .await
+        .unwrap();
+
+    if let MaybePendingTransactionReceipt::Receipt(TransactionReceipt::Invoke(InvokeTransactionReceipt {
+        events,
+        ..
+    })) = deployment_of_counter_evm_contract_result_receipt
+    {
+        events.iter().find_map(|event| {
+            if event.keys.contains(&get_selector_from_name("evm_contract_deployed").unwrap()) {
+                Some(event.data.clone())
+            } else {
+                None
+            }
+        })
+    } else {
+        None
+    }
+}
+
+async fn deploy_starknet_contract(
+    account: &SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>,
+    class_hash: &FieldElement,
+    constructor_calldata: Vec<FieldElement>,
+) -> Result<FieldElement, Box<dyn std::error::Error>> {
+    let factory = ContractFactory::new(*class_hash, account);
+
+    let result = factory.deploy(constructor_calldata.clone(), FieldElement::ZERO, false).send().await.unwrap();
+
+    let contract_address =
+        get_contract_address(FieldElement::ZERO, *class_hash, &constructor_calldata.clone(), FieldElement::ZERO);
+
+    // TODO: add a check here
+    let _receipt = account.provider().get_transaction_receipt(result.transaction_hash).await.unwrap();
+
+    Ok(contract_address)
+}
+
+async fn declare_kkrt_contracts(
+    account: &SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>,
+) -> HashMap<String, FieldElement> {
+    let paths = fs::read_dir("tests/compiled_kkrt").expect("Could not read directory");
```

Comment:
`tests/compiled_kakarot`

i'd suggest putting the paths in module constants (with const keyword, &str, and fullcaps)